superset/common/query_context.py

```diff
@@ -92,19 +92,22 @@ def __init__(
         self,
         datasource: DatasourceDict,
         queries: List[Dict[str, Any]],
-        force: bool = False,
-        custom_cache_timeout: Optional[int] = None,
         result_type: Optional[ChartDataResultType] = None,
         result_format: Optional[ChartDataResultFormat] = None,
+        force: bool = False,
+        custom_cache_timeout: Optional[int] = None,
     ) -> None:
         self.datasource = ConnectorRegistry.get_datasource(
             str(datasource["type"]), int(datasource["id"]), db.session
         )
-        self.force = force
-        self.custom_cache_timeout = custom_cache_timeout
         self.result_type = result_type or ChartDataResultType.FULL
         self.result_format = result_format or ChartDataResultFormat.JSON
-        self.queries = [QueryObject(self, **query_obj) for query_obj in queries]
+        for qd in queries:
+            qd.setdefault("result_type", result_type)
+        query_object_factory = QueryObjectFactory()
+        self.queries = [query_object_factory.create(**query_obj) for query_obj in queries]
```

Comment:
```Python
       self.queries = [QueryObjectFactory.create(**query_obj) for query_obj in queries]
```

Reply:
1. In general, using class methods should be avoided, hard to test.
2. this is only a temp change, after that PR will be another PRs so the Factory will move to another module and config will be injected to it so it wont be decouple to superset

Reply:
@zhaoyongjie look at #17479

Reply:
Thanks! I will take a closer look tomorrow.